CloudyCider/models/EC2PageListener.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  EC2PageListener.swift
+//  CloudyCider
+//
+//  Created by Matvii Hodovaniuk on 7/7/19.
+//  Copyright © 2019 Matvii Hodovaniuk. All rights reserved.
+//
+
+import AWSSDKSwiftCore
+import EC2
+import Foundation
+
+final class EC2PageListener: EC2Listener {
+  func loadPage(store: EC2Store) {
+    // MARK: - EC2Init
+
+    let accessKeyId = ProcessInfo.processInfo.environment["accessKeyId"]
+    let secretAccessKey = ProcessInfo.processInfo.environment["secretAccessKey"]
+    let ec2 = EC2(
+      accessKeyId: accessKeyId,
+      secretAccessKey: secretAccessKey,
+      region: .euwest2
+    )
+    let instanceRequest = EC2.DescribeInstancesRequest()
+    do {
+      try ec2.describeInstances(instanceRequest).whenComplete { response in
+        do {
+          let result = try response.get()
+          guard let reservations = result.reservations else { return }
+
+          // create temp storage for the instances list
+          var nextStoreInstances: [EC2Instance] = []
+
+          for reservation in reservations {
+            guard let instances = reservation.instances else { return }
+
+            for instance in instances {
+              // get the instance state
+              guard let state = instance.state,
+                let statusName = state.name else { return }
+
+              // get the instance name
+              guard let tags = instance.tags else { return }
+              let nameTag = tags.filter { $0.key == "Name" }[0]
```

Comment:
this bakes in an assumption that `tags` is a non-empty array, better use `first` instead of `[0]`